### ISSUE TITLE
Add temporary on/off block option for tests

### ISSUE THREAD

**Issue by veganstraightedge** [hoverinc/this_feature#8]
We could turn a feature on/off in before/after blocks, respectively.

```ruby
before { FFeature::Test.flag(:foobarbaz).on }
after { FFeature::Test.flag(:foobarbaz).off }
```

And/or we could use a block to wrap a test. Something like this:

```ruby
it "does something when the Foo Bar Baz feature is on" do
  Feature::Test.(:foobarbaz).on do
    # test setup
    # test change
    # test expectation
  end
end
```

**Comment by maxp-hover** [comment]
We have changed this since Matt's last update. 

Test adapter is now defined through configuration:

```
# config/initializers/this_feature.rb
require "this_feature"
require "this_feature/adapters/memory"
require "this_feature/adapters/flipper"

ThisFeature.configure do |config|
  case Rails.env
  when "test"
    config.test_adapter = ThisFeature::Adapters::Memory.new(context_key_method: :this_feature_id)
    config.adapters = [config.test_adapter]
  else
    config.adapters = [ThisFeature::Adapters::Flipper.new]
  end
end
```

And used in tests via the `ThisFeature.test_adapter` getter:

```
before { ThisFeature.test_adapter.on!(:job_address_check, context: job.org) }
```

My plan for now is to add the following to `spec_helper.rb` in hyperion:

```
  config.after(:each) do
    ThisFeature.test_adapter.clear
  end
```

That way each flag will get automatically set to "off" after each run, so we don't need to do any cleanup on a per-test basis.

**Comment by maxp-hover** [comment]
I think we can merge this issue once https://github.com/hoverinc/hyperion/pull/7117 is merged, since we have a separate issue to discuss adding `on!` and `off!` at the top level.
